### PR TITLE
change(cred-libs): Update user_password to username_password

### PR DESCRIPTION
**Pull request by louisruch**
Creating a new cred library with user_password fails:
```
boundary credential-libraries create vault -credential-store-id $CRED_STORE_ID -vault-path "secret/data/my-secret" -name "Super secret lib" -credential-type user_password
Error from controller when performing create on vault-type credential library

Error information:
  Kind:                InvalidArgument
  Message:             Error in provided request.
  Status:              400
  context:             Error from controller when performing create on vault-type credential library

  Field-specific Errors:
    Name:              -credential-type
      Error:           Unknown credential type "user_password"
```

Creating a new cred library with username_password:
``` 
boundary credential-libraries create vault -credential-store-id $CRED_STORE_ID -vault-path "secret/data/my-secret" -name "Super secret lib" -credential-type username_password

Credential Library information:
  Created Time:          Wed, 08 Jun 2022 15:19:20 PDT
  Credential Store ID:   csvlt_VlFgkW6EGX
  ID:                    clvlt_gJUfsRyyFR
  Name:                  Super secret lib
  Type:                  vault
  Updated Time:          Wed, 08 Jun 2022 15:19:20 PDT
  Version:               1

  Scope:
    ID:                  p_1234567890
    Name:                Generated project scope
    Parent Scope ID:     o_1234567890
    Type:                project

  Authorized Actions:
    no-op
    read
    update
    delete

  Attributes:
    HTTP Method:         GET
    Path:                secret/data/my-secret

  Credential Type:
    username_password
```

Connecting to target:
```
boundary connect -target-id ttcp_1234567890

Proxy listening information:
  Address:             127.0.0.1
  Connection Limit:    -1
  Expiration:          Wed, 08 Jun 2022 23:20:29 PDT
  Port:                50861
  Protocol:            tcp
  Session ID:          s_dkx9C3MJ8L

  Credentials:
    Credential Source ID:   clvlt_gJUfsRyyFR
    Credential Source Name: Super secret lib
    Credential Store ID:    csvlt_VlFgkW6EGX
    Credential Store Type:  vault
    Credential Type:        username_password
    Secret:
          password:   pass
          username:   test
```

Curl string output
```
boundary credential-libraries create vault -credential-store-id $CRED_STORE_ID -vault-path "secret/data/my-secret" -name "Super secret lib" -credential-type username_password -output-curl-string

curl -X POST -H "Authorization: Bearer $(boundary config get-token -keyring-type keychain -token-name default)" -H "Content-Type: application/json" -d '{"attributes":{"path":"secret/data/my-secret"},"credential_store_id":"csvlt_VlFgkW6EGX","credential_type":"username_password","name":"Super secret lib"}' 'http://127.0.0.1:9200/v1/credential-libraries'
```

### CODE REVIEW

**Review by jefferai — APPROVED** [review]

**Comment by louisruch** [comment]
@ddebko PTAL

**Review by ddebko — APPROVED** [review]
LGTM

File: internal/db/schema/migrations/oss/postgres/31/01_credential_type.up.sql [hunk 1]
```diff
@@ -0,0 +1,22 @@
+begin;
+
+  -- drop constraint so we can migrate user_password to username_password
```
Comment (tmessi) [inline internal/db/schema/migrations/oss/postgres/31/01_credential_type.up.sql:3]: can you add a comment to note that this replaces the constraint defined in `2/02_credential_type.up.sql` and likewise add a comment in `2/02_credential_type.up.sql` to note that the constraint is replaced in `31/01_credential_type.up.sql`

**Review by tmessi — CHANGES_REQUESTED** [review]
I think your commit prefix should be `refact` not `change`.

**Review by tmessi — APPROVED** [review]

**Pull request merged**